uplinks/freifunk-berlin-uplink-tunnelberlin-tunneldigger-files/uci-defaults/freifunk-berlin-z95_tunnelberlin-tunneldigger

```diff
@@ -1,24 +1,33 @@
 #!/bin/sh
 
-. /lib/functions/system.sh
+THIS_UPLINKNAME="tunnelberlin_tunnneldigger"
+
+. /lib/functions/freifunk-berlin-network.sh
+. /lib/functions.sh
 
 # always set correct masquerading, regardless of guard
 uci set firewall.zone_ffuplink.masq=1
 uci commit firewall
 
-uci -q get ffberlin-uplink || echo "" | uci import ffberlin-uplink
-uci >/dev/null -q get ffberlin-uplink.preset || uci set ffberlin-uplink.preset=settings
-uci >/dev/null -q get ffberlin-uplink.preset.current || uci set ffberlin-uplink.preset.current="tunnelberlin_tunneldigger"
-if [[ $(uci get ffberlin-uplink.preset.current) != '"tunnelberlin_tunneldigger"' ]]; then
-  uci rename ffberlin-uplink.preset.current=previous
-  uci set ffberlin-uplink.preset.current="tunnelberlin_tunneldigger"
+current_preset=$(uci get ffberlin-uplink.preset.current)
+if [ ${current_preset} != ${THIS_UPLINKNAME} ]; then
+  if [ ${current_preset} != "undefined" ]; then
+      # when the uplink-preset has changed, recreate remember the preset we are coming from
+      # and prepare for reinit of the relevant settings via freifunk-berlin-ffuplink-defaults
+      # uci-defaults
+      logger -t "ffuplink" "uplink-preset has been changed."
+      uci rename ffberlin-uplink.preset.current=previous
+      create_ffuplink
+  fi
+  uci set ffberlin-uplink.preset.current=${THIS_UPLINKNAME}
 fi
 # set set auth-type required for this uplink-type, e.g. for freifunk-wizard
 uci set ffberlin-uplink.uplink=settings
 uci set ffberlin-uplink.uplink.auth=none
+
 uci commit ffberlin-uplink
 
-. /lib/functions/guard.sh
+, /lib/functions/guard.sh
```

Comment:
this looks wrong, should be "." right?

Reply:
You are right.  Commit adb6c3d0c295994acaf0b705358b8db728e54b48 has been added to for-Hedy-1.0.x_change_uplink